gcm_filters/kernels.py

```diff
@@ -290,6 +299,115 @@ def __call__(self, field: ArrayType):
 ALL_KERNELS[GridType.TRIPOLAR_POP_WITH_LAND] = POPTripolarLaplacianTpoint
 
 
+@dataclass
+class VectorLaplacian(BaseLaplacian):
+    """̵Vector Laplacian on C-Grid.
+
+    Attributes
+    ----------
+    wet_mask_t: Mask array for t points, 1 for ocean, 0 for land
+    wet_mask_q: Mask array for q (vorticity) points, 1 for ocean, 0 for land
+    dxT: x-spacing centered at t points
+    dyT: y-spacing centered at t points
+    dxCu: x-spacing centered at u points
+    dyCu: y-spacing centered at u points
+    dxCv: x-spacing centered at v points
+    dyCv: y-spacing centered at v points
+    dxBu: x-spacing centered at q points
+    dyBu: y-spacing centered at q points
+    area_u: U-cell area
+    area_v: V-cell area
+    kappa_iso: isotropic viscosity
+    kappa_aniso: anisotropic viscosity aligned with x-direction
+    """
+
+    wet_mask_t: ArrayType
+    wet_mask_q: ArrayType
+    dxT: ArrayType
+    dyT: ArrayType
+    dxCu: ArrayType
+    dyCu: ArrayType
+    dxCv: ArrayType
+    dyCv: ArrayType
+    dxBu: ArrayType
+    dyBu: ArrayType
+    area_u: ArrayType
+    area_v: ArrayType
+    kappa_iso: ArrayType
+    kappa_aniso: ArrayType
+
+    def __post_init__(self):
+        np = get_array_module(self.wet_mask_t)
+
+        self.dx_dyT = self.dxT / self.dyT * self.wet_mask_t
+        self.dy_dxT = self.dyT / self.dxT * self.wet_mask_t
+        self.dx_dyBu = self.dxBu / self.dyBu * self.wet_mask_q
+        self.dy_dxBu = self.dyBu / self.dxBu * self.wet_mask_q
+
+        self.dx2h = self.dxT * self.dxT
+        self.dy2h = self.dyT * self.dyT
+        self.dx2q = self.dxBu * self.dxBu
+        self.dy2q = self.dyBu * self.dyBu
+
+        # compute reciprocal of areas while avoiding division by zero
+        self.recip_area_u = np.where(self.area_u > 0, 1 / self.area_u, 0)
+        self.recip_area_v = np.where(self.area_v > 0, 1 / self.area_v, 0)
+
+    def __call__(self, ufield: ArrayType, vfield: ArrayType):
```

Comment:
My concern here is the the signature of this Laplacian does not match the signature of `BaseLaplacian`, which is

```python
def __call__(self, field):
```

I'm surprised that mypy did not complain about that. In any case, it supports the idea suggested above that we should have a separate base class for vector laplacians.

Reply:
Good point. Definitely a good idea to have two separate base classes.